src/goose/brokers/redis/commands.clj

```diff
@@ -251,11 +251,33 @@
   (wcar* conn (car/zremrangebyscore sorted-set sorted-set-min score)))
 
 ;;; ============ Hashes ============
-
 (defn parse-map [conn hash]
   (wcar* conn (car/parse-map (car/hgetall hash) :keywordize)))
 
 ;;; ============ Misc ============
-
 (defn exists [conn key]
   (wcar* conn (car/exists key)))
+
+;;; ============ Batch ============
+(defn get-batch-state
+  [conn id]
+  (let [batch-state-key (d/prefix-batch id)
+        enqueued-job-set (d/construct-batch-job-set id d/enqueued-job-set)
+        retrying-job-set (d/construct-batch-job-set id d/retrying-job-set)
+        successful-job-set (d/construct-batch-job-set id d/successful-job-set)
+        dead-job-set (d/construct-batch-job-set id d/dead-job-set)
+        [_ [batch-state enqueued retrying successful dead]]
+        (car/atomic conn
+                    atomic-lock-attempts
+                    (car/multi)
+                    (car/hgetall batch-state-key)
+                    (car/scard enqueued-job-set)
+                    (car/scard retrying-job-set)
+                    (car/scard successful-job-set)
+                    (car/scard dead-job-set))]
```

Comment:
question: should we take into into account that one of these commands as a part of the transaction might fail?

Reply:
The transaction would fail if one of the commands fails, causing the fn call to return an error. 

However, Redis doesn't fail the entire transaction if we perform an operation on a key with wrong value ([ref](https://redis.io/docs/interact/transactions/#errors-inside-a-transaction)). 
Since Goose's implementation creates these keys, we won't run into such an error unless Redis keys are directly manipulated by the user.
And such an error wouldn't occur